### ISSUE TITLE
405 error for /?wc-ajax=get_refreshed_fragments ajax callbacks with Cache enabler method 

### ISSUE THREAD

**Issue by colmtroy** [centminmod/centminmod#142]
**Background Information**

Please fill in any relevant information that applies to you:

* **CentOS Version:** i.e. CentOS 6 32bit or 64bit / CentOS 7 64bit ?
* **Centmin Mod Version Installed:** i.e. 123.08stable or 123.09beta01
* **Nginx Version Installed:** nginx/1.19.4 (131120-001910-centos7-ba36c7f)
* **PHP Version Installed:** PHP 7.4.12 (cli) (built: Nov 13 2020 00:23:02) ( NTS )
* **MariaDB MySQL Version Installed:** 10.3.27-MariaDB
* **When was last time updated Centmin Mod code base ? :** Nov 14 2020
* **Persistent Config:** 
`LETSENCRYPT_DETECT='y'`
* **Install Method:** How was Centmin Mod installed ? Exact specific steps or commands used.
`yum -y update; curl -O https://centminmod.com/betainstaller74.sh && chmod 0700 betainstaller74.sh && bash betainstaller74.sh`
**Describe the bug**
Installed a new version of WP with the Cache Enabler cache method.
Installed WooCommerce and added some products.
Open site with dev tools network tab open ( e.g. https://ovh3.commercegurus.com/ )
Filter requests by XHR requests.
Observe /?wc-ajax=get_refreshed_fragments is a 405
https://i.imgur.com/aMV3q95.png

**Expected behavior**
Request should return a 200 response

**Additional context**
Issue doesn't exist with Redis cache option

**Comment by centminmod** [comment]
which version of cache enabler are you using 1.4.9 or 1.5.1+. If you're using 1.5.1+, tried downgrading to cache enabler 1.4.9 https://community.centminmod.com/threads/add-cache-enabler-1-4-9-legacy-cache-mode-option-in-123-09beta01.20545/ ?

cache enabler 1.5.1+ changed the way they cache for query strings https://github.com/keycdn/cache-enabler/issues/148 so you might want to check with cache enabler support too to see if other folks reporting similar issues at https://wordpress.org/support/plugin/cache-enabler/

**Comment by centminmod** [comment]
also does manually clearing cache enabler cache, fix the ?wc-ajax requests temporarily ?

**Comment by colmtroy** [comment]
hey @centminmod 
Sorry I should have led with that info!
I'm running 1.4.9 of Cache Enabler.
I spotted your threads on the big changes to Cache Enabler when I was updating another WP install last week (sidnote: I think the change to caching query strings by default is not going to end well! So many plugins rely on query strings!).
Clearing cache doesn't solve it - it's been like that since Day 1. (this is a brand new machine I just setup over the weekend).

**Comment by colmtroy** [comment]
hey @centminmod 
just a quick update on this.
created another brand new machine following the steps listed above and this is still an issue.

**Comment by centminmod** [comment]
@colmtroy I'll take a look but I don't have much experience with Woocommerce myself.

**Comment by colmtroy** [comment]
@centminmod 
i think i've figured it out.
I added wc-ajax to the bypass rules in:
wpcacheenabler_mydomain.com.conf

```bash
    # bypass cache if URLs containing the following strings
    if ($request_uri ~* "(\?add-to-cart=|\?wc-ajax=|\?wc-api=|/cart/|/my-account/|/checkout/|/shop/checkout/|/wp-json/|/store/checkout/|/customer-dashboard/|/addons/|/wp-admin/|/xmlrpc.php|/wp-(app|cron|login|register|mail).php|wp-.*.php|/feed/|index.php|wp-comments-popup.php|wp-links-opml.php|wp-locations.php|sitemap(_index)?.xml|[a-z0-9_-]+-sitemap([0-9]+)?.xml)") {
        set $cache_uri 'nullcache';
    }
```
not sure why this rule is needed given you already have this a bit further up

```bash
    # bypass cache if POST requests or URLs with a query string
    if ($request_method = POST) {
        set $cache_uri 'nullcache';
    }
```

and the request in question is a post request
```
698:188.141.52.86 - - [04/May/2021:13:24:03 +0000] "POST /?wc-ajax=get_refreshed_fragments HTTP/2.0" 405 552 "https://snip/product/shorts-with-a-tie-belt/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.85 Safari/537.36"
```

**Comment by centminmod** [comment]
Nice troubleshooting! Could be the order of the rules so will add wc-ajax to bypass $request_uri rule too :)

**Comment by centminmod** [comment]
oh i checked in latest wpcacheenabler_domain.com.conf templates, the POST request bypass is higher up before the bypass URL rules so shouldn't be the case - if you did fresh install it would of been the same case

**Comment by centminmod** [comment]
added fix to 123.09beta01 https://github.com/centminmod/centminmod/commit/d29de854098ce89116e86fd1de2626924549c9fd :)

**Comment by colmtroy** [comment]
Awesome thanks as always @centminmod :)
